Update editor.d.ts to v10.276

Spice-King opened the pull request:
Fixes #2088, now depends on #2104.

kmoschcau commented on src/foundry/client/ui/editor.d.ts:
```diff
@@ -1,5 +1,5 @@
 import type { ConfiguredDocumentClassForName } from '../../../types/helperTypes';
```
I think this needs to be extended to also support prosemirror. Have a look at #2095 's `FormApplication#activateEditor`. It effectively returns the value of `TextEditor.create` and is marked to return either `tinymce.Editor` or one of prosemirror's `EditorView`s.

Spice-King replied:
Right, going to make this depend on #2104 then, as I pull in the types for it there.

Spice-King commented:
Intentionally a slight bit broken as it now also depends on #2109 for CONFIG changes.

kmoschcau commented:
Once the merge conflicts here are solved and the diff updated, I can review it.

Spice-King commented:
I've rebased it, since I forgot to redo it after I fixed the prosemirror one. I'll try to avoid needing to stack PRs in the future, and try to curb the habit of trying to keep a clean history with rebasing, amends and what not.
